client-src/elements/chromedash-guide-editall-page.js

```diff
@@ -157,24 +137,76 @@ export class ChromedashGuideEditallPage extends LitElement {
     `;
   }
 
+  findProcessStage(feStage) {
+    for (const processStage of this.process.stages) {
+      if (feStage.stage_type === processStage.stage_type) {
+        return processStage;
+      }
+    }
+    return null;
+  }
+
+  getStageFormFields(processStage) {
+    return FLAT_FORMS_BY_INTENT_TYPE[processStage.outgoing_stage] || [];
+  }
+
+  renderStageFormFields(formattedFeature, processStage, feStage, formFields) {
+    if (!formFields) return nothing;
+
+    return html`
+      <h3>${processStage.name}</h3>
+      <section class="flat_form">
+        ${formFields.map((field) => html`
+          <chromedash-form-field
+            name=${field}
+            stageId=${feStage.stage_id}
+            value=${formattedFeature[field]}>
+          </chromedash-form-field>
+        `)}
+      </section>
+    `;
+  }
+
+  getForms(formattedFeature, feStages) {
+    const formsToRender = [html`
+    <h3>Feature metadata</h3>
+    <section class="flat_form">
+      ${METADATA_FORM_FIELDS.map((field) => html`
+        <chromedash-form-field
+          name=${field}
+          value=${formattedFeature[field]}>
+        </chromedash-form-field>
+      `)}
+    </section>
+    `];
+
+    let allFormFields = [...METADATA_FORM_FIELDS];
+    for (const feStage of feStages) {
+      const processStage = this.findProcessStage(feStage);
+      if (!processStage) {
+        continue;
+      }
+      const formFields = this.getStageFormFields(processStage);
+      formsToRender.push(this.renderStageFormFields(
+        formattedFeature, processStage, feStage, formFields));
+      allFormFields = [...allFormFields, ...formFields];
+    }
+
+    return [allFormFields, formsToRender];
+  }
+
   renderForm() {
+    const formattedFeature = formatFeatureForEdit(this.feature);
```

Comment:
Could we use the instance variable `this.featureForEdit` here?   It just seems like two ways to get the same result, so we should pick one.